vispy/visuals/multichannel_image.py

```diff
@@ -0,0 +1,413 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+"""Image visual constructed from 3 separate image arrays."""
+
+import warnings
+
+import numpy as np
+from ..gloo.texture import should_cast_to_f32
+from ._scalable_textures import GPUScaledTexture2D
+from .image import ImageVisual
+from .shaders import Function, FunctionChain
+
+
+class MultiChannelGPUScaledTexture2D:
+    """Wrapper class around individual textures.
+
+    This helper class allows for easier handling of multiple textures that
+    represent individual R, G, and B channels of an image.
+
+    """
+
+    _singular_texture_class = GPUScaledTexture2D
+    _ndim = 2
+
+    def __init__(self, data, **texture_kwargs):
+        # data to send to texture when not being used
+        self._fill_arr = np.full((10, 10), np.float32(np.nan),
+                                 dtype=np.float32)
+
+        self.num_channels = len(data)
+        data = [x if x is not None else self._fill_arr for x in data]
+        self._textures = self._create_textures(self.num_channels, data,
+                                               **texture_kwargs)
+
+    def _create_textures(self, num_channels, data, **texture_kwargs):
+        return [
+            self._singular_texture_class(data[i], **texture_kwargs)
+            for i in range(num_channels)
+        ]
+
+    @property
+    def textures(self):
+        return self._textures
+
+    @property
+    def clim(self):
+        """Get color limits used when rendering the image (cmin, cmax)."""
+        return tuple(t.clim for t in self._textures)
+
+    def set_clim(self, clim):
+        if isinstance(clim, str) or len(clim) == 2:
```

Comment:
nit: call this `clims` like you do in the other clim setter makes this a bit easier to read.